### PR TITLE
Added Jreleaser

### PR DESCRIPTION
**Pull request by mburkert3**
Added JReleaser so we can publish directly to maven central

It relates to the following issue #s:

[BNR-1226](https://sonatype.atlassian.net/browse/BNR-1226)

cc @bhamail / @DarthHater / @guillermo-varela / @shaikhu

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -26,5 +26,10 @@ jobs:
           fetch-depth: 0
       - uses: ./.github/actions/setup
       - uses: ./.github/actions/configure-git
-      - name: Release with Gradle
-        run: ./gradlew clean release -Dorg.gradle.daemon=false --stacktrace -x test -x integrationTest
+      - name: Release with JReleaser
+        run: ./gradlew clean release -Dorg.gradle.daemon=false --stacktrace
```
Comment (shaikhu) [inline .github/workflows/release.yml:30]: IIUC the release is run after a successful CI build, so there isn't a need to run all the integration tests again.

File: build.gradle [hunk 2]
```diff
@@ -86,110 +87,80 @@ test {
...
+      artifactId = project.name
+      version = project.version.toString()
+
+      pom {
+        name = project.name
+        description = pluginDescription
+        url = 'https://github.com/sonatype-nexus-community/scan-gradle-plugin'
+        organization {
+          name = 'Sonatype'
+          url = 'https://www.sonatype.org'
```
Comment (shaikhu) [inline build.gradle:103]: Doesn't load for me. However `https://www.sonatype.com` does.

**Comment by shaikhu** [comment]
Question. JReleaser is for releasing to central right? The existing logic is for releasing to the gradle plugin portal?

File: .github/workflows/release.yml (hunk 1 above)
Comment (guillermo-varela) [reply .github/workflows/release.yml:30]: > there isn't a need to run all the integration tests again.

Agreed

File: build.gradle [hunk 3]
```diff
@@ -86,110 +87,80 @@ test {
...
-    def pluginUrl = 'https://github.com/sonatype-nexus-community/scan-gradle-plugin'
-
-    pluginBundle {
-      website = pluginUrl
-      vcsUrl = pluginUrl
-      description = pluginDescription
-      tags = ['sonatype', 'scan', 'dependencies', 'ossindex', 'iq server']
-      System.setProperty('gradle.publish.key', System.getenv('GRADLE_PUBLISH_KEY') ?: '')
-      System.setProperty('gradle.publish.secret', System.getenv('GRADLE_PUBLISH_SECRET') ?: '')
-    }
```
Comment (guillermo-varela) [inline build.gradle:99]: We need to keep this part so the plugin is also listed in the Gradle plugins portal:
https://plugins.gradle.org/plugin/org.sonatype.gradle.plugins.scan

File: build.gradle [hunk 4]
```diff
@@ -86,110 +87,80 @@ test {
...
     }
   }
 }
 
+// Configure signing
+signing {
+  useInMemoryPgpKeys(System.getenv('GPG_SIGNING_KEY'), System.getenv('GPG_PASSPHRASE'))
+  sign publishing.publications.maven
+
+}
```
Comment (guillermo-varela) [inline build.gradle:161]: Would this attempt to sign the compiled binary in local development as well?

File: build.gradle [hunk 5]
```diff
@@ -86,110 +87,80 @@ test {
...
+
+
 afterReleaseBuild.dependsOn publish
 publish.finalizedBy closeAndReleaseRepository
```
Comment (guillermo-varela) [inline build.gradle:165]: This is possibly no longer needed as I believe is from the old publishing plugin

Comment (mburkert3) [reply build.gradle]: [Commit](https://github.com/sonatype-nexus-community/scan-gradle-plugin/pull/174/commits/905e23895bc5553b1a239aa45cd852585ad6d599)

File: .github/workflows/release.yml (hunk 1 above)
Comment (shaikhu) [reply .github/workflows/release.yml:30]: @mburkert3 I think the `-x test -x integrationTest` needs to be added back?

File: build.gradle [hunk 6]
```diff
@@ -86,110 +87,90 @@ test {
...
+          stagingRepository('build/staging-deploy')
+        }
       }
     }
   }
 }
 
+// Configure signing
+signing {
+  useInMemoryPgpKeys(System.getenv('GPG_SIGNING_KEY'), System.getenv('GPG_PASSPHRASE'))
```
Comment (bhamail) [inline build.gradle:170]: Will this handle decoding the base64 encoded PGP signing key?

**Review by bhamail — APPROVED** [review]
+1, Looks good to me. Just curious to make sure the Base84 decode of the GPG key is handled somewhere, as the secret is base64 encoded.

File: build.gradle (hunk 6 above)
Comment (mburkert3) [reply build.gradle:170]: I believe the useInMemoryPGPKeys should handle that

File: build.gradle [hunk 7]
```diff
@@ -86,110 +87,99 @@ test {
...
+        }
       }
     }
   }
 }
 
+if (System.getenv('RELEASE_BUILD') == 'true') {
+  signing {
+    def final encodedKey = System.getenv('GPG_SIGNING_KEY')
+    def final signingKey = new String(encodedKey.decodeBase64())
```
Comment (bhamail) [inline build.gradle:171]: +1

**Review by bhamail — APPROVED** [review]
+1

**Pull request merged**